src/agent/stacktrace-parser/src/asan.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::StackEntry;
+use anyhow::{bail, Result};
+use regex::Regex;
+
+pub(crate) fn parse_asan_call_stack(text: &str) -> Result<Vec<StackEntry>> {
```

Comment:
Is this code ported, or ours? (In the future, I'd love us to move away from regex-based parsing)

Reply:
This is mine, written from scratch, but thematically similar.  The libclusterfuzz regex akin to this one is extremely unergonomic to use and update.

Reply:
And yes, I'd like to move away from regexes as well, but... as with everything: time.